### PR TITLE
[feature-wip] (datetimev2) support window funnel and modify valid dat…

### PR DESCRIPTION
**Pull request by Gabriel39**
# Proposed changes

Issue Number: close #xxx

## Checklist(Required)

1. Type of your changes:
    - [ ] Improvement
    - [ ] Fix
    - [x] Feature-WIP
    - [ ] Feature
    - [ ] Doc
    - [ ] Refator
    - [ ] Others: 
2. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
3. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
5. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
6. Are there any changes that cannot be rolled back:
    - [ ] Yes
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/catalog/FunctionSet.java [hunk 1]
```diff
@@ -1388,6 +1388,19 @@ private void initAggregateBuiltins() {
                 "",
                 "",
                 true, false, true, true));
+        addBuiltin(AggregateFunction.createBuiltin(FunctionSet.WINDOW_FUNNEL,
```
Comment (BiteTheDDDDt) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/FunctionSet.java:1391]: Does we need add `DATEV2` override?

Comment (Gabriel39) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/FunctionSet.java:1391]: `window_funnel` will be searched in `IS_NONSTRICT_SUPERTYPE_OF` mode so DATEV2 could be cast to DATETIMEV2 implicitly

Comment (BiteTheDDDDt) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/FunctionSet.java:1391]: Ok, then do `AggregateFunctionWindowFunnel<DateV2Value<DateV2ValueType>, UInt32>` still need to register?

Comment (Gabriel39) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/FunctionSet.java:1391]: Only `create_aggregate_function_window_funnel` is registered in function factory as a function creator. When we pass a window argument with date type, a function implement `AggregateFunctionWindowFunnel<DateV2Value<DateV2ValueType>, UInt32>` will be returned. So no extra things is registered

**Review by BiteTheDDDDt — DISMISSED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**